### PR TITLE
Can't just make checkboxes and let'em unchecked [solved]

### PR DESCRIPTION
**Pull request by brunovcosta**
It is possible use select tags in formFields

### CODE REVIEW

**Comment by taromero** [comment]
Hey thanks for the PR!

Before reviewing it, could you please expand a bit more about the issue this is solving? It's not be clear to me at the moment, even after reading the #5 ticket.

**Comment by brunovcosta** [comment]
Hello, 
In fact tis PR was made to solve another issue i had (use select/option). As it was my first PR on github, i tried to solve an easier problem. I'm haven't figured out how solve the #5 ticket.

File: live-demo/live-demo.html [hunk 1]
```diff
@@ -16,4 +16,4 @@
   <button id='sample1'>Sample 1</button>
   <button id='complex'>Complex</button>
 </body>
-</html>
```
Comment (taromero) [inline live-demo/live-demo.html:19]: Please avoid introducing unnecessary changes to files.

File: swal-forms.css [hunk 2]
```diff
@@ -1,7 +1,7 @@
 .swal-form {
   padding-top: 3%;
-  overflow: scroll;
-  height: 300px;
+  overflow: auto;
```
Comment (taromero) [inline swal-forms.css:3]: How does this change affect the layout? Could you attach a screenshot?

File: swal-forms.js [hunk 3]
```diff
@@ -46,10 +46,13 @@
     addWayToGetFormValuesInDoneFunction: function(swalArgs) {
       var swalFormInstance = this
       var doneFunction = swalArgs[1]
-      swalArgs[1] = function() {
+      swalArgs[1] = function(isConfirm) {
```
Comment (taromero) [inline swal-forms.js:50]: I think this newly introduced `isConfirm` parameter used?

File: swal-forms.js [hunk 4]
```diff
@@ -59,7 +62,7 @@
       return inputArray
               .filter(uncheckedRadiosAndCheckboxes)
               .map(toValuableAttrs)
-              .reduce(toSingleObject)
+              .reduce(toSingleObject,{})
```
Comment (taromero) [inline swal-forms.js]: I don't see how adding this `{}` would add value. Input for the `reduce` call should be either an empty array or an array filled with objects. This might solve an issue, but it's not clear to me which one.

File: swal-forms.js (hunk 3 above)
Comment (brunovcosta) [inline swal-forms.js:50]: I used this parameter to force the doneFunction to be called. Doing this, I could bring the flow of application to force erase the innerHtml even if swal were canceled

File: swal-forms.js [hunk 5]
```diff
@@ -160,13 +163,27 @@
...
-        return t("<input id='{id}' class='{clazz}' type='{type}' name='{name}'" +
-                  " value='{value}' title='{placeholder}' placeholder='{placeholder}'>" + 
-                "<label for='{name}'>{label}</label>", input)
+        if (input.type!=="select"){
+            var inputTag = t("<input id='{id}' class='{clazz}' type='{type}' name='{name}'" +
+                 " value='{value}' title='{placeholder}' placeholder='{placeholder}'>", input);
+        }else{
+            var inputTag = t("<select id='{id}' class='{clazz}' name='{name}'" +
+                 " value='{value}' title='{placeholder}' style='width:100%'>", input);
+            input.options.forEach(function(option){
```
Comment (taromero) [inline swal-forms.js:200]: I like the overall approach to add <select> capabilities, though this particular call could be replaced buy a `reduce` call (since we are basically generating a single string by transforming a set of items).

File: swal-forms.css (hunk 2 above)
Comment (brunovcosta) [inline swal-forms.css:3]: this change didn't interfere in swal-forms mechanics, it just made the alert windows look nicier(in my opinion). So, you can ignore this.

Here is the screenshot:
image

File: live-demo/live-demo.html (hunk 1 above)
Comment (brunovcosta) [inline live-demo/live-demo.html:19]: Sorry about that. It was my mistake.

**Comment by taromero** [comment]
Hi!

I've left some comments on the PR that would need either correction or clarification in order to merge it.

Here are a couple of additional changes that this PR needs in order for me to merge it:
1. PR's title describes a problem, rather than a change that fixes or adds something. The latter should be preferred.
2. It seems that this PR provides multiple unrelated features/fixes. It would be better to split each feature/fix into a separate PR. Ideally separate issues describing problems or possible improvements and separate PRs linked to them.
3. I've just added [feross/standard](https://github.com/feross/standard) compliance to the project. This adds some styling rules for new code. Please rebase against master and make sure that the code comply to the feross/standard rules.

This are rules I'd like to enforce on my repositories to make contributions easier to manage.

File: swal-forms.js (hunk 3 above)
Comment (taromero) [inline swal-forms.js:50]: Could you link to a line in the code where this parameter is used? I can't find it.

Comment (brunovcosta) [inline swal-forms.js:50]: it is not used in this code. when you pass an function with no params to sweetAlert, this is called only if you click in the ok button. If this function is declared with one param, this function is called when the swal is canceled too.
image

Comment (taromero) [inline swal-forms.js:50]: I believe there might be a misunderstanding with the sweet-alertexample you link to. The way how I understand it works is that the same callback function is called regardless of which button ("ok" or "cancel") is clicked, and this `isConfirm` parameter adds the possibility to handle each scenario. In the example you link, the parameter is used for this purpose, while in the code where you are introducing this `isConfirm` parameter it is not used at all.

Comment (brunovcosta) [inline swal-forms.js:50]: I believe this is the way I said before. You can check this by open your console(f12 in google chrome or firefox). then:
`swal({title: "teste"},function(){alert("hello")})`
Press Esc key.
Now type this:
`swal({title: "teste"},function(someArgument){alert("hello")})`
Press Esc key again

Comment (taromero) [inline swal-forms.js:50]: I see the same modal and behaviors with the 2 swal calls described in the comment above. On both cases a swal modal is displayed with just an "OK" button which when clicked opens an alert. Do you observe a different behavior?

Comment (brunovcosta) [inline swal-forms.js:50]: the diferent behavior came from Pressing Esc key to cancel the sweetAlert, instead of clicking in OK Button.

[link to test](https://jsfiddle.net/tfxjx47q/)

Comment (taromero) [inline swal-forms.js:50]: I observed the difference in the behavior now. Thanks.
